lib/server_connection.ml

```diff
@@ -155,6 +155,7 @@ let error_code t =
   else None
 
 let shutdown t =
+  Queue.clear t.request_queue;
```

Comment:
If there is an active connection, this will cause any active request handlers to hang on reading from the request body.

Reply:
Yeah I wasn't thrilled by this. Definitely worth revisiting and writing more tests for.

Reply:
This was removed. See last commit for details.